lib/include/jxl/gain_map.h

```diff
@@ -0,0 +1,110 @@
+/* Copyright (c) the JPEG XL Project Authors. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file.
+ */
+
+/** @addtogroup libjxl_metadata
+ * @{
+ * @file gain_map.h
+ * @brief Utility functions to manipulate jhgm (gain map) boxes.
+ */
+
+#ifndef JXL_GAIN_MAP_H_
+#define JXL_GAIN_MAP_H_
+
+#include <jxl/color_encoding.h>
+#include <jxl/jxl_export.h>
+#include <jxl/types.h>
+
+#ifndef __cplusplus
+extern "C" {
+#endif
+
+/** Gain map bundle */
+typedef struct {
+  /** version number */
+  uint8_t jhgm_version;
+  /** size of the gain map metadata */
+  uint16_t gain_map_metadata_size;
+  /** pointer to binary blob of gain map metadata (ISO 21496-1) */
+  const uint8_t* gain_map_metadata;
+  /** indicate if it has a color encoding*/
+  bool has_color_encoding;
+  /** uncompressed color encoding */
+  JxlColorEncoding color_encoding;
+  /** size of the alt_icc profile  (compressed size) */
+  uint32_t alt_icc_size;
+  /** pointer to the compressed icc profile */
+  const uint8_t* alt_icc;
+  /** size of the gain map */
+  uint32_t gain_map_size;
+  /** pointer to the gain map (a JPEG XL naked codestream) */
+  const uint8_t* gain_map;
+} JxlGainMapBundle;
+
+/**
+ * Calculates the total size required to serialize the gain map bundle into a
+ * binary buffer. This function accounts for all the necessary space to
+ * serialize fields such as gain map metadata, color encoding, compressed ICC
+ * profile data, and the gain map itself.
+ *
+ * @param[in] map_bundle Pointer to the JxlGainMapBundle containing all
+ * necessary data to compute the size.
+ * @param[out] bundle_size The size in bytes required to serialize the bundle.
+ * @return Whether setting the size was successful.
+ */
+JXL_EXPORT JXL_BOOL JxlGainMapGetBundleSize(const JxlGainMapBundle* map_bundle,
```

Comment:
Do we expect this function to be able to fail in the future?

Reply:
It can already fail now, no?